src/axom/sidre/core/MFEMSidreDataCollection.cpp

```diff
@@ -1287,6 +1386,189 @@ std::string MFEMSidreDataCollection::getElementName(mfem::Element::Type elementE
   return "unknown";
 }
 
+mfem::Geometry::Type MFEMSidreDataCollection::getElementTypeFromName(
+  const std::string& name)
+{
+  if(name == "point")
+    return mfem::Geometry::POINT;
+  else if(name == "line")
+    return mfem::Geometry::SEGMENT;
+  else if(name == "tri")
+    return mfem::Geometry::TRIANGLE;
+  else if(name == "quad")
+    return mfem::Geometry::SQUARE;
+  else if(name == "tet")
+    return mfem::Geometry::TETRAHEDRON;
+  else if(name == "hex")
+    return mfem::Geometry::CUBE;
+  else
+    return mfem::Geometry::INVALID;
+}
+
+// private method
+void MFEMSidreDataCollection::reconstructMesh()
+{
+  // mfem::ConduitDataCollection::BlueprintMeshToMesh would be useful here, but
+  // we need all the parameters to construct a ParMesh (the Mesh base subobject
+  // is initialized manually)
+
+  SLIC_ERROR_IF(
+    !verifyMeshBlueprint(),
+    "Cannot reconstruct mesh, data does not satisfy Conduit Blueprint");
+
+  SLIC_ERROR_IF(!m_bp_grp->hasView("coordsets/coords/values/x"),
```

Comment:
I think this function will need some additional work (perhaps this is what you meant by "experimental").

When reading in a mesh, the coordinate set does not need to be named `coords`.
Similarly, the topology does not need to be named 'mesh'
And it will not always have a "boundary" topology (even if some ranks have a "boudary topology -- we've been burned by that one in the past).

Reply:
Perhaps the claim is that this function is only guaranteed/expected to `Load` a blueprint that was `Save`d by this class, `MFEMSidreDataCollection`, in which case some of the coord/topology names can be reasonably hard-coded?

Reply:
Yes, reconstruction is currently only supported for meshes/fields written by this class.  This requirement will be stricter for parallel meshes as the shared edge/face data is currently "off-label use" of the Conduit blueprint.

Reply:
As a follow-up, I have added a snippet to the documentation that specifies the required topology/coordset names, and also converted these to variables instead of having the paths hardcoded.  There are still some places that use the hardcoded names, so a follow-up PR will address this.  Having a `Path` class (see #438) will make this easier and more readable.